specification/dashboard/resource-manager/Microsoft.Dashboard/preview/2021-09-01-preview/grafana.json

```diff
@@ -0,0 +1,748 @@
+          "$ref": "#/definitions/ManagedIdentity"
+        },
+        "systemData": {
+          "description": "The system meta data relating to this grafana resource.",
+          "$ref": "#/definitions/SystemData"
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "location": {
+          "type": "string"
+        }
+      }
+    },
+    "GrafanaResourceListResponse": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/GrafanaResource"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "GrafanaResourceProperties": {
+      "type": "object",
+      "description": "Properties specific to the grafana resource.",
+      "properties": {
+        "provisioningState": {
+          "description": "Provisioning state of the resource.",
+          "$ref": "#/definitions/ProvisioningState"
+        },
+        "grafanaVersion": {
+          "description": "The Grafana software version.",
+          "type": "string",
+          "readOnly": true
+        },
+        "endpoint": {
+          "description": "The endpoint of the Grafana instance.",
+          "type": "string",
+          "readOnly": true
+        },
+        "zoneRedundancy": {
+          "$ref": "#/definitions/ZoneRedundancy"
+        }
+      }
+    },
+    "GrafanaResourcePropertiesUpdateParameters": {
+      "type": "object",
+      "properties": {
+        "zoneRedundancy": {
+          "description": "The high availability mode of the grafana resource.",
+          "$ref": "#/definitions/ZoneRedundancy"
+        }
+      }
+    },
+    "GrafanaResourceUpdateParameters": {
+      "type": "object",
+      "description": "The parameters for a PATCH request to a grafana resource.",
+      "properties": {
+        "identity": {
+          "description": "The managed identity of the grafana resource.",
+          "$ref": "#/definitions/ManagedIdentity"
+        },
+        "tags": {
+          "type": "object",
+          "description": "The new tags of the grafana resource.",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "location": {
+          "description": "The location of the grafana resource.",
+          "type": "string"
+        },
+        "properties": {
+          "description": "Properties specific to the grafana resource.",
+          "$ref": "#/definitions/GrafanaResourcePropertiesUpdateParameters"
+        }
+      }
+    },
+    "LastModifiedByType": {
+      "enum": [
+        "User",
+        "Application",
+        "ManagedIdentity",
+        "Key"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "LastModifiedByType"
+      }
+    },
+    "ManagedIdentity": {
```

Comment:
Use common types as explained above

Reply:
Hi @raosuhas , after comparing it with https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/types.json#L253, I think we have followed the structure of this object in common-types.

As there are other properties (like `None` in the `type` and `userAssignedIdentities`), we are not able to directly refer to the common types.